ubccr/xdmod
# Update sso user retrieval

## Description
Added additional functionality to both SSO & normal login path that makes sure
that the user that is authenticating has an up to date organization_id. A user's
organization can become out of date if the person they are associated with has
their organization changed. It can also occur if a User's person is updated but not their organization. 

We also do not check the organization if the user has been assigned the unknown
person or organizations.

Added a new `MailHelper` class that's a thin wrapper around
`MailWrapper::sendMail` to catch / log any exceptions that may be encountered
while sending an email.

## Motivation and Context
We need to be able to account for when a user's assigned person has their organization updated. 

## Tests performed
Manual Tests Performed:

- Manually update a centerdirector's user's person to somebody with a different organization.
- Login as the centerdirector
- Check that email(s) have been sent
- Check that the Users record has been updated w/ the new person & organization
- Check that the user no longer has the center director acl. 
- Create an SSO User
- Give them the center director acl 
- Give them a person w/ an organization != Unknown Organization
- Login via SSO
- Ensure that no email has been generated
- Manually update this users's person w/ somebody who has a different organization
- Login via SSO
- Ensure that emails are generated
- Ensure that the user has their person / organization updated
- Ensure that the user no longer has the center director acl

## Types of changes


## Checklist:
